### ISSUE TITLE
The provider still shows infura endpoint even setting walletconnect/web3-provider custom rpc to BSC chain

### ISSUE THREAD

**Issue by happyeric77** [Uniswap/web3-react#211]
Hi team, 
I am making a DAPP on BSC which allows users to access through browser and mobile. So I would like to integrate the WalletConnect into my DAPP.

I simply install the npm package "@walletconnect/web3-provider". with the following code:

```
import WalletConnectProvider from "@walletconnect/web3-provider"
try {
    console.log("Log in with connect wallet")
    const provider = new WalletConnectProvider({
        rpc: {
            56: "https://bsc-dataseed1.binance.org",
        },
    });
    await provider.enable();
    const web3 = new Web3(provider);
    resolve(web3)
} catch {
    console.log("Install metamask or use wallet connect")
} 
```
It seems to work normally in the beginning because the QR code pops up and I can use my trust wallet to scan and connect it. 
And then I found out it seems not to able to send transactions with the following error message on the console:
```
[Error] Failed to load resource: the server responded with a status of 401 ()  https://mainnet.infura.io/v3(v3, line 0)
```
And I tried to print out the web3 and checked the provider object. and it still shows that the provider connection is under "mainnet.infura.io/v3"

```
http: HTTPConnection {_events: Object, _eventsCount: 2, url: "https://mainnet.infura.io/v3/", eventNames: function, listeners: function, …}
```

It will be appreciated if anyone teaches me how to make it work.

**Comment by happyeric77** [comment]
I have managed to solve it out, see below link:
https://github.com/WalletConnect/walletconnect-monorepo/issues/522#issuecomment-860773569

**Comment by h0tw4t3r** [comment]
@happyeric77 would you please mind reopening it? Me and other people still having issues as stated in https://github.com/WalletConnect/walletconnect-monorepo/issues/522

**Comment by grabbou** [comment]
FYI that was fixed just now in #777. I reproduced this and resolved. Just reopened for this satisfaction of closing once again! Please upgrade to latest WalletConnect connector.
